### PR TITLE
add respondent address lookup, manual and select screens feature

### PR DESCRIPTION
**Pull request by giorgi-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRLC100-35

### Change description ###
- Added address/lookup screen
- Added address/manual screen
- Added address/select screen
- address postControllers now use URL params 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by giorgi-hmcts** [comment]
There's an issue with `emailAddress` and `telephoneNumber` not being saved in the userCase from the respondent `contact-details` page.

File: src/main/steps/c100-rebuild/respondent-details/address/lookup/content.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+export const generateFormFields = (caseData: Partial<C100RebuildPartyDetails>): GenerateDynamicFormFields => {
+  return { fields: lookupAddressForm(caseData).fields, errors: { en: {}, cy: {} } };
+};
+
+export const generateContent: TranslationFn = content => {
+  const translations = languages[content.language]();
+  const lookupAddressFormTranslations = lookupAddressFormLanguages[content.language]();
+  const respondentId = content?.additionalData?.req?.params!.respondentId as C100RebuildPartyDetails['id'];
+  const respondentData = content.userCase?.resp_Respondents!.find(
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/respondent-details/address/lookup/content.ts:67]: Can we please use common util method getRespndentDetails

File: src/main/steps/c100-rebuild/respondent-details/address/manual/ManualAddressPostController.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const { saveAndSignOut, saveBeforeSessionTimeout, _csrf, ...formData } = form.getParsedBody(req.body);
+
+    const respondentId1: AnyType | undefined = respondentId;
+
+    const respondent = this.getRespondentDetails(
+      //eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+      req.session.userCase?.resp_Respondents!,
+      respondentId1
+    ) as C100RebuildPartyDetails;
+    respondent!.address!.PostCode = req.body['addressPostcode'] as string;
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/respondent-details/address/manual/ManualAddressPostController.ts:31]: We can use spread operator here I think to reduce these manual assignment.
@vivek-sekhar Can you please add your input ?

File: src/main/steps/c100-rebuild/respondent-details/address/manual/ManualAddressPostController.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    respondent!.address!.AddressLine2 = req.body['address2'] as string;
+    respondent!.address!.PostTown = req.body['addressTown'] as string;
+    respondent!.address!.County = req.body['addressCounty'] as string;
+    respondent!.address!.addressHistory = req.body['addressHistory'] as YesOrNo;
+    respondent!.address!.provideDetailsOfPreviousAddresses = req.body['provideDetailsOfPreviousAddresses'] as string;
+
+    req.session.errors = form.getErrors(formData);
+    this.redirect(req, res);
+  }
+  private getRespondentDetails = (
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/respondent-details/address/manual/ManualAddressPostController.ts:42]: This is not required, we can use common util getRespndentDetails method please

File: src/main/steps/c100-rebuild/respondent-details/address/manual/content.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+export const generateFormFields = (caseData: Partial<C100RebuildPartyDetails>): GenerateDynamicFormFields => {
+  return { fields: manualAddressForm(caseData).fields, errors: { en: {}, cy: {} } };
+};
+
+export const generateContent: TranslationFn = content => {
+  const translations = languages[content.language]();
+  const manualAddressFormTranslations = manualAddressFormLanguages[content.language]();
+  const respondentId = content?.additionalData?.req?.params!.respondentId;
+  const respondentData = content.userCase?.resp_Respondents!.find(
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/respondent-details/address/manual/content.ts:93]: Please use common util getRespndentDetails method

File: src/main/steps/c100-rebuild/respondent-details/address/select/SelectAddressPostController.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        //eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const selectedAddress = req.session.addresses[selectedAddressIndex] as any;
+
+        const respondentId1: AnyType | undefined = respondentId;
+        const respondent = this.getRespondentDetails(
+          //eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+          req.session.userCase?.resp_Respondents!,
+          respondentId1
+        ) as C100RebuildPartyDetails;
+        (respondent!.address!.PostCode = selectedAddress.postcode as string),
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/respondent-details/address/select/SelectAddressPostController.ts:37]: Same as above comment. We can use spread operator I think

File: src/main/steps/c100-rebuild/respondent-details/address/select/SelectAddressPostController.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          (formData['AddressLine1'] = selectedAddress.street1);
+        formData['AddressLine2'] = selectedAddress.street2;
+        formData['PostTown'] = selectedAddress.town;
+        formData['County'] = selectedAddress.county;
+        formData['PostCode'] = selectedAddress.postcode;
+      }
+    }
+    this.redirect(req, res);
+  }
+  private getRespondentDetails = (
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/respondent-details/address/select/SelectAddressPostController.ts:52]: Can we please use common util getRespndentDetails method

File: src/main/steps/c100-rebuild/respondent-details/address/select/content.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+export const generateFormFields = (caseData: Partial<C100RebuildPartyDetails>): GenerateDynamicFormFields => {
+  return { fields: selectAddressForm(caseData).fields, errors: { en: {}, cy: {} } };
+};
+
+export const generateContent: TranslationFn = content => {
+  const translations = languages[content.language]();
+  const selectAddressFormTranslations = selectAddressFormLanguages[content.language](content);
+  const respondentId = content?.additionalData?.req?.params!.respondentId;
+  const respondentData = content.userCase?.resp_Respondents!.find(
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/respondent-details/address/select/content.ts:67]: Let's use util getRespndentDetails method please

**Review by channa-patil — CHANGES_REQUESTED** [review]

**Review by channa-patil — APPROVED** [review]
TODO: Need to take care of these comments later in the base branch once all people section changes merged.

**Comment by giorgi-hmcts** [comment]
Sure @channa-patil. 
Noted to take care of these comments once all people changes are merged.

**Pull request merged**